integration/spark/src/main/common/java/io/openlineage/spark/agent/lifecycle/plan/SqlDWDatabricksVisitor.java

```diff
@@ -0,0 +1,125 @@
+package io.openlineage.spark.agent.lifecycle.plan;
+
+import io.openlineage.client.OpenLineage;
+import io.openlineage.spark.api.DatasetFactory;
+import io.openlineage.spark.api.OpenLineageContext;
+import io.openlineage.spark.api.QueryPlanVisitor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Supplier;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.apache.commons.lang3.reflect.FieldUtils;
+import org.apache.commons.lang3.reflect.MethodUtils;
+import org.apache.spark.sql.catalyst.plans.logical.LogicalPlan;
+import org.apache.spark.sql.execution.datasources.LogicalRelation;
+import org.apache.spark.sql.sources.BaseRelation;
+
+/**
+ * {@link LogicalPlan} visitor that matches SqlDWRelation that comes from an Azure Databricks
+ * environment. This function extracts a {@link OpenLineage.Dataset} from the SQL DW/ Synapse table
+ * referenced by the relation. The convention used for a namespace is a URI of <code>
+ * sqlserver://&lt;server&gt;.&lt;.datasetId&gt;.&lt;tableName&gt;</code> . The name for Sql Dw
+ * tables may be a query (e.g. <code>
+ * ((select \"id\" FROM dbo.exampleInputA WHERE postalCode != '55555') q)</code>) or a table name
+ * (e.g. "exampleInputA") or a multi-part name (e.g. "[dbo].[exampleInputA]").
+ */
+public class SqlDWDatabricksVisitor<D extends OpenLineage.Dataset>
+    extends QueryPlanVisitor<LogicalPlan, D> {
+  private final DatasetFactory<D> factory;
+  private static final Pattern dbJdbcPattern = Pattern.compile("database=([^;]*);?");
+  private static final Pattern serverJdbcPattern = Pattern.compile("jdbc:([^;]*);?");
+
+  public SqlDWDatabricksVisitor(OpenLineageContext context, DatasetFactory<D> factory) {
+    super(context);
+    this.factory = factory;
+  }
+
+  public static boolean hasSqlDWDatabricksClasses() {
+    try {
+      SqlDWDatabricksVisitor.class
+          .getClassLoader()
+          .loadClass("com.databricks.spark.sqldw.SqlDWRelation");
+      return true;
+    } catch (Exception e) {
+      // swallow- we don't care
+    }
+    return false;
+  }
+
+  @Override
+  public boolean isDefinedAt(LogicalPlan plan) {
+    return sqlDwSupplier(plan).isPresent();
+  }
+
+  private Optional<Supplier<BaseRelation>> sqlDwSupplier(LogicalPlan plan) {
+    if (!(plan instanceof LogicalRelation)) {
+      return Optional.empty();
+    }
+
+    Class relationClass = ((LogicalRelation) plan).relation().getClass();
+    if (plan instanceof LogicalRelation
+        && relationClass.toString().equals("class com.databricks.spark.sqldw.SqlDWRelation")) {
```

Comment:
This should be `relationClass.getName().equals("com.databricks.spark.sqldw.SqlDWRelation")`

Reply:
If you extract this line into a `protected` method, you can write tests for this code. e.g., 

```java
protected boolean isSqlDwRelationClass(LogicalRelation plan) {
  return plan instanceof LogicalRelation && 
    ((LogicalRelation) plan).relation().getClass().getName().equals("com.databricks.spark.sqldw.SqlDWRelation");
}
```
allows you to write a test class like

```java
class TestSqlDwDatabricksVisitor extends SqlDwDatabricksVisitor {
  @Override
  protected boolean isSqlDwRelationClass(LogicalRelation plan) {
    return true;
  }
}
```
Include a test `BaseRelation` that looks something like the `SqlDwRelation` will allow you to test the regexes and the reflection code.

Reply:
Thank you for the guidance, Mike!  I was able to create a couple tests based off of this insight.